### PR TITLE
Adds support for ES 7.8.0

### PR DESCRIPTION
**Pull request by skkosuri-amzn**
*Issue #, if available:*
https://github.com/opendistro-for-elasticsearch/alerting/issues/218

*Description of changes:*
- Bump ODFE version to 1.9.0
- Bump Elasticsearch version to 7.8.0
- Bump Gradle version to 6.5
- Changes to Alerting plugin code to make it compatible with Elasticsearch 7.8.0 data structures. 
- Fix jacocoagent path to work with gradle 6.5 for integration tests.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: notification/build.gradle [hunk 1]
```diff
@@ -18,6 +18,7 @@ apply plugin: 'java'
 apply plugin: 'jacoco'
 apply plugin: 'maven-publish'
 apply plugin: 'signing'
+apply plugin: 'elasticsearch.testclusters'
```
Comment (bowenlan-amzn) [inline notification/build.gradle:21]: do we need this to apply here?

Comment (skkosuri-amzn) [reply notification/build.gradle:21]: good point, we dont need that. Will remove.

File: alerting/build.gradle [hunk 2]
```diff
@@ -81,8 +81,10 @@ dependencies {
 }
 
 javadoc.enabled = false // turn off javadoc as it barfs on Kotlin code
-licenseHeaders.enabled = true
+licenseHeaders.enabled = false
```
Comment (dbbaughe) [inline alerting/build.gradle:84]: Was this breaking something? Or why did we change to false here, seems like licenseHeaders check won't be run now?

File: alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/AlertingPlugin.kt [hunk 3]
```diff
@@ -131,7 +132,8 @@ internal class AlertingPlugin : PainlessExtension, ActionPlugin, ScriptPlugin, P
         environment: Environment,
         nodeEnvironment: NodeEnvironment,
         namedWriteableRegistry: NamedWriteableRegistry,
-        indexNameExpressionResolver: IndexNameExpressionResolver
+        indexNameExpressionResolver: IndexNameExpressionResolver,
+        repositoriesServiceSupplier: Supplier<RepositoriesService>?
```
Comment (dbbaughe) [inline alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/AlertingPlugin.kt:136]: Is this actually nullable or just added from the autocomplete?

File: build.gradle [hunk 4]
```diff
@@ -62,6 +62,8 @@ allprojects {
     }
 }
 
+
```
Comment (dbbaughe) [inline build.gradle:65]: np: remove extra newlines

File: notification/build.gradle [hunk 5]
```diff
@@ -45,6 +46,8 @@ task javadocJar(type: Jar) {
     from javadoc.destinationDir
 }
 
+
```
Comment (dbbaughe) [inline notification/build.gradle:49]: np: extra newlines?

File: gradle/wrapper/gradle-wrapper.properties [hunk 6]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.4-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-bin.zip
```
Comment (dbbaughe) [inline gradle/wrapper/gradle-wrapper.properties:3]: Did anything change in the wrapper for 6.4 -> 6.5? Don't see anything checked in PR
https://docs.gradle.org/current/userguide/gradle_wrapper.html#sec:upgrading_wrapper

File: alerting/build.gradle (hunk 2 above)
Comment (skkosuri-amzn) [reply alerting/build.gradle:84]: Did this change from job-schedular plugin change. I guess we don't need it. will check and change accordingly.

File: alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/AlertingPlugin.kt (hunk 3 above)
Comment (skkosuri-amzn) [reply alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/AlertingPlugin.kt:136]: Autocomplete.

File: build.gradle (hunk 4 above)
Comment (skkosuri-amzn) [reply build.gradle:65]: ack

File: notification/build.gradle (hunk 5 above)
Comment (skkosuri-amzn) [reply notification/build.gradle:49]: ack

File: gradle/wrapper/gradle-wrapper.properties (hunk 6 above)
Comment (ftianli-amzn) [reply gradle/wrapper/gradle-wrapper.properties:3]: I checked by running `./gradlew wrapper --gradle-version 6.5` command to upgrade the wrapper, and nothing got changed.

File: alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/AlertingPlugin.kt (hunk 3 above)
Comment (ftianli-amzn) [reply alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/AlertingPlugin.kt:136]: look forward to seeing what will be changed here, if something need to be changed.  :smile:

Comment (dbbaughe) [reply alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/AlertingPlugin.kt:136]: Ok, can we remove the ? at the end then as it isn't actually nullable

Comment (skkosuri-amzn) [reply alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/AlertingPlugin.kt]: Done.

**Pull request merged**